solr/core/src/java/org/apache/solr/servlet/SolrDispatchFilter.java

```diff
@@ -398,32 +397,12 @@ public CoreContainer getCores() {
   
   @Override
   public void destroy() {
-    if (closeOnDestroy) {
-      close();
-    }
-  }
-  
-  public void close() {
-    CoreContainer cc = cores;
-    cores = null;
     try {
-      if (metricManager != null) {
-        try {
-          metricManager.unregisterGauges(registryName, metricTag);
-        } catch (NullPointerException e) {
-          // okay
-        } catch (Exception e) {
-          log.warn("Exception closing FileCleaningTracker", e);
-        } finally {
-          metricManager = null;
-        }
-      }
-    } finally {
-      if (cc != null) {
-        httpClient = null;
-        cc.shutdown();
-      }
+      cores.getMetricManager().unregisterGauges(registryName, metricTag);
```

Comment:
I wonder about this - perhaps useful in tests, but pretty slow and seemingly almost always unnecessary in real life. But unrelated.

Reply:
Didn't seem slow when I looked at the code some weeks ago.  If I'm wrong, then I'd propose that our tests have you opt-in to such so you don't pay for features you don't plan to use.